src/lib/zf/cplane.c

```diff
@@ -182,7 +182,9 @@ __zf_cplane_get_path(struct zf_stack* st, struct zf_path* path,
   struct cp_fwd_data data;
   cicp_verinfo_t verinfo;
   int rc;
-  int iter;
+  time_t time_limit_us;
+  time_t time_used_us = 0;
```

Comment:
I realise @abower-amd asked for `time_t` to be used, but is it safe to cast a signed integer to `time_t` given it isn't guaranteed to be signed (although is in seemingly most cases)? I expect this would only be very problematic for an unsigned 64-bit `time_t` (if one even exists), as a 32-bit unsigned `time_t` would only wait for a little over 4 seconds.

Reply:
Yes, there could be a problem if it's an unsigned type (unlikely) and `gettimeofday` jumps backwards (theoretically possible) giving a negative value. Better to use an explicitly signed type (long or int64_t) and/or use `clock_gettime(CLOCK_MONOTONIC)` to avoid negative times.

Reply:
`time_t` is defined to be signed but yes the key thing is for it to be a 64-bit signed quantity however that is achieved and I agree CLOCK_MONOTONIC is the right way to do this.

Reply:
`time_t` is implementation-defined, but we're quite strongly tied to a particular implementation so I suppose it doesn't really matter if we assume the definition is suitable for this useage. I'd still use an explicitly signed and sized type if that's what I wanted.

Reply:
I think my concern was more with the possibility of a negative value being provided for `sti->arp_reply_timeout` which, when cast to `time_t`, could result in a massive `time_limit_us`, potentially many years long in the worst case of an unsigned implementation.

Reply:
Sorry Mike, yes you are right the signedness isn't guaranteed. I agree this should simply be `int64_t` and `CLOCK_MONOTONIC` should be used.